Packs/SailPointIdentityNow/Integrations/SailPointIdentityNow/SailPointIdentityNow.py

```diff
@@ -0,0 +1,622 @@
+
+# Other constants
+# Using beta account API for accounts
+SEARCHABLE_TYPE_LIST = ['identities', 'accessprofiles', 'roles', 'entitlements', 'events']
+MAX_INCIDENTS_TO_FETCH = 250
+MAX_LIMIT = 250
+OFFSET_DEFAULT = '0'
+LIMIT_DEFAULT = '250'
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%S'
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    """
+    Client class to interact with API request.
+    """
+
+    def __init__(self, base_url: str, verify: bool, proxy: bool, headers: dict, max_results: int, request_timeout: int):
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy, headers=headers)
+        self.max_results = max_results
+        self.request_timeout = request_timeout
+
+    def send_request(self, url_suffix: str, method: str, params=None, json_data=None):
+        """
+        Perform a HTTP request to IdentityNow API.
+
+        :type url_suffix: ``str``
+        :param url_suffix: IdentityNow API endpoint ext suffix.
+
+        :type method: ``str``
+        :param method: HTTP method, e.g. 'GET', 'POST', 'PUT', 'DELETE'.
+
+        :type params: ``JSON``
+        :param params: URL parameters to specify the query.
+
+        :type json_data: ``JSON``
+        :param json_data: Data to be sent as part of 'POST' or 'PUT' request.
+
+        :return: Response after fulfilling the request successfully, else None.
+        """
+        if url_suffix is None or method is None:
+            return None
+
+        return self._http_request(url_suffix=url_suffix, method=method, json_data=json_data, params=params,
+                                  timeout=self.request_timeout, resp_type='response',
+                                  ok_codes=(200, 201, 202, 204, 400, 401, 403, 404, 429, 500), proxies=handle_proxy())
+
+
+''' HELPER/UTILITY FUNCTIONS '''
+
+
+def get_headers(base_url: str, client_id: str, client_secret: str, grant_type: str):
+    """
+    Create header with OAuth 2.0 authentication information.
+
+    :type base_url: ``str``
+    :param base_url: Base URL of the IdentityNow tenant.
+
+    :type client_id: ``str``
+    :param client_id: Client Id for OAuth 2.0.
+
+    :type client_secret: ``str``
+    :param client_secret: Client Secret for OAuth 2.0.
+
+    :type grant_type: ``str``
+    :param grant_type: Grant Type for OAuth 2.0. Defaulted to 'client_credentials' if not provided.
+
+    :return: Header with OAuth 2.0 information if client_id & client_secret are provided, else None.
+    This will return None if the client_id & client_secret were not valid (authorized).
+    """
+    if base_url is None or client_id is None or client_secret is None:
+        return None
+
+    if grant_type is None:
+        grant_type = 'client_credentials'
+
+    params = {
+        'grant_type': grant_type,
+        'client_id': client_id,
+        'client_secret': client_secret
+    }
+    oauth_response = requests.request("POST", url=f'{base_url}{IDN_OAUTH_EXT}', params=params)
```

Comment:
Why not use the client's http request?

Reply:
The clients http request needs the headers which are request by this API call. This is also used for performing a test connection, so wanted to keep the client functionality out of this.